stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/ChannelNameFormatter.kt

```diff
@@ -0,0 +1,68 @@
+package io.getstream.chat.android.ui.common
+
+import android.content.Context
+import androidx.annotation.StringRes
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.common.extensions.getDisplayName
+
+/**
+ *  An interface that allows to generate a name for the given channel.
+ */
+public fun interface ChannelNameFormatter {
+
+    /**
+     * Generates a name for the given channel.
+     *
+     * @param channel The channel whose data is used to generate the name.
+     * @param currentUser The currently logged in user.
+     * @return The display name for the given channel.
+     */
+    public fun formatChannelName(channel: Channel, currentUser: User?): String
+
+    public companion object {
+        /**
+         * Build the default channel name formatter.
+         *
+         * @param context The context to load string resources.
+         * @param fallback The resource identifier of a fallback string.
+         *
+         * @see [DefaultChannelNameFormatter]
+         */
+        public fun defaultFormatter(
+            context: Context,
+            @StringRes fallback: Int = R.string.stream_ui_channel_list_untitled_channel,
+        ): ChannelNameFormatter {
+            return DefaultChannelNameFormatter(context, fallback)
+        }
+    }
+}
+
+/**
+ * A simple implementation of [ChannelNameFormatter] that allows to generate a name for a channel
+ * based on the following rules:
+ *
+ * - If the channel has a name, then its name is returned
+ * - If the channel is distinct, then a comma-separated list of member names is returned
+ * - Otherwise, the placeholder text defined in [R.string.stream_ui_channel_list_untitled_channel] is returned
+ *
+ * @param context The context to load string resources.
+ * @param fallback The resource identifier of a fallback string.
+ */
+private class DefaultChannelNameFormatter(
+    private val context: Context,
+    @StringRes private val fallback: Int,
+) : ChannelNameFormatter {
+
+    /**
+     * Generates a name for the given channel.
+     *
+     * @param channel The channel whose data is used to generate the name.
+     * @param currentUser The currently logged in user.
+     * @return The display name for the given channel.
+     */
+    override fun formatChannelName(channel: Channel, currentUser: User?): String {
+        return channel.getDisplayName(context, currentUser, fallback)
```

Comment:
Is our default now the same as the iOS default?

@adasiewiczr I think we decided to align this with the iOS team so that we can show the same information.

Reply:
For convenience, posting Swift docs here:

```
/// - If channel has some name, ie. `Channel 1`, this returns `Channel 1`
/// - If channel has no name and is not direct message, this returns channel ID of the channel
/// - If channel is direct message, has no name and has members where there just 2,
/// returns name of the members in alphabetic order: `Leia, Luke`
/// - If channel is direct message, has no name and has members where there are more than 2,
/// returns name of the members in alphabetic order with how many members left: `Leia, Luke and 5 others`
///  - If channel is direct message, has no name and no members, this returns `nil`
///  - If channel is direct message, has no name and only one member, shows the one member name
///
/// - Parameters:
///   - maxMemberNames: Maximum number of visible members in Channel defaults to `2`
///   - separator: Separator of the members, defaults to `y`
/// - Returns: A closure with 2 parameters carrying `channel` used for name generation and `currentUserId` to decide
/// which members' names are going to be displayed
```

Reply:
Decided to go with the default Android - if someone wants custom formatting, they can override it.